web/admin/src/Admin.js

```diff
@@ -69,34 +72,37 @@ export default class Admin extends PureComponent {
         { sessionId && <div>
             <label className="row">
               <span>Session Name:&nbsp;</span>
-              <input type="text" value={sessions[sessionId].name} onChange={this.onSessionNameChange} />
+              <input type="text" value={sessions[sessionId].name} maxLength={50} onChange={this.onSessionNameChange} />
+              {sessions[sessionId].isDisplayable ? null : <p>Please Rename Session</p>}
```

Comment:
`isDisplayable` is another new variable that can be derived from others and that can get out of sync.  If the name is empty, we don't need an extra `isDisplayable` flag... we know something is wrong with it.  If it is a duplicate, how do we determine which session's name should be changed?  I think it is fine to alert the CMS user that a name is duplicated (and this is derivable just by looking at the names of all the sessions, without a separate flag) but I would not add this flag and get the client side code in the business of hiding such sessions.

Reply:
Yes the purpose of this was two fold to alert the admin that a session is a duplicate or blank and hide it from the user. Should we add a pending state to handle incomplete questions or sessions?

Reply:
I would avoid having a flag to signal that a question or session is incomplete. We should be able to tell just by looking at its fields when we want to know.  Having a separate field that tells whether it's incomplete duplicates data, and those duplicates can get out of sync.